flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java

```diff
@@ -486,20 +485,22 @@ private void registerTableSinkInternal(String name, TableSink<?> tableSink) {
 					// wrapper contains only sink (not source)
 					ConnectorCatalogTable sourceAndSink = ConnectorCatalogTable
 						.sourceAndSink(sourceSinkTable.getTableSource().get(), tableSink, !IS_STREAM_TABLE);
-					catalogManager.alterTable(sourceAndSink, getTemporaryObjectIdentifier(name), false);
```

Comment:
`alterTable` is not used anymore, right?

Reply:
Correct, alterTable works with permanent table. We do not have an API for that.